### PR TITLE
target/linux: add missing patch headers

### PR DESCRIPTION
**Pull request by feckert**
I am currently working with the external kernel tree 5.15 to add a new target.
So I don't have to work with quilt I decided to make the integration with an external git tree.
I wanted to apply all the patches from the 5.15 kernel in openwrt to the external kernel tree.
With the sequence [backport-15.5](https://github.com/openwrt/openwrt/tree/master/target/linux/generic/backport-5.15), [pending-15.5](https://github.com/openwrt/openwrt/tree/master/target/linux/generic/pending-5.15) and [hack-15.5](https://github.com/openwrt/openwrt/tree/master/target/linux/generic/hack-5.15).
During apply in my external git tree with `git am <openwrt>/target/linux/generic/pending-15.5/*` and `get am <openwrt>/target/linux/hack-15.5/*` I noticed that some patches do not have a valid header and therefore cannot be applied easily.

@Ansuel @hauke could you please have a look?

Also, from my point of view, it makes sense to describe in the patch body what this patch does. I have also seen that this should normaly be done as requested by this file https://github.com/openwrt/openwrt/blob/master/target/linux/generic/PATCHES.

As a non-author of the patch, it would be easier for me and the others, if there was more information in the body about why this change is necessary. Then maybe someone else could send this patch upstream to get this into the kernel. Also a rebasing of the patch set could also be done easier.

### CODE REVIEW

**Comment by BKPepe** [comment]
>Also, from my point of view, it makes sense to describe in the patch body what this patch does. I

Yes! :100: 

A similar thing is that generally, patches should be submitted first to the upstream repository instead of having them in the tree forever. See: https://lists.openwrt.org/pipermail/openwrt-adm/2022-March/002221.html

**Comment by Ansuel** [comment]
Yhea the hack series is the worst on this kind of thing... Also related to
the fact that we have patch that are present from the born of openwrt
itself.



Il Mer 13 Lug 2022, 14:29 Florian Eckert ***@***.***> ha
scritto:

**Comment by Ansuel** [comment]
@feckert i'm just checking them... can you refresh the patch? I think the git version at the end of the patch will be stripped by quilt refresh

**Comment by feckert** [comment]
Thanks for feedback.

> @feckert i'm just checking them... can you refresh the patch? I think the git version at the end of the patch will be stripped by quilt refresh

I will do this asap.

However, if this is accepted, I will be listed as the author!
Should I do a git history search who is the author of the this patches I am changing?

**Comment by Ansuel** [comment]
@feckert that's a lot of work but yes that would be really good... also considering some of them are never pushed and are very trivial like some of them add just an id for a usb device...

File: target/linux/generic/hack-5.15/261-lib-arc4-unhide.patch [hunk 1]
```diff
@@ -1,7 +1,14 @@
-This makes it possible to select CONFIG_CRYPTO_LIB_ARC4 directly. We 
-need this to be able to compile this into the kernel and make use of it 
-from backports.
```
Comment (Ansuel) [inline target/linux/generic/hack-5.15/261-lib-arc4-unhide.patch:3]: @feckert think we have a commit description here... somehow

**Comment by feckert** [comment]
@Ansuel I have now searched to the best of my knowledge the git history for the pending patches that do not have a valid header and added a Signed-off-by where it was clear and updated the FROM of the patch. And I also added a message body where it was possible and clear what the patch was doing

But I'm not sure if I'm allowed to add the Signed-off-by at all!
From my point of view, only the FROM should be added in the patch.

I will do the hacks tomorrow.

**Comment by Ansuel** [comment]
@feckert mhhh adding from without sob is a no go for anyone that wants to upstream them.... 

well think we are a bit stuck about it... We should ask to the single FROM user if we can add the sob and wait some days for a response...

then with the ones we have a response we add a sob (and ideally i can try to push them upstream) and for the other leave the from tag and fix your problem. Thanks for the good work!

**Comment by tiagogaspar8** [comment]
>But I'm not sure if I'm allowed to add the Signed-off-by at all!

Can't the signed off by be obtained through the commiter in git history? That way it would be accurate and true/not fabricated

**Comment by tiagogaspar8** [comment]
I see what you mean now, the person could've obtained the code from another place 😅

**Comment by Ansuel** [comment]
@tiagogaspar8 or he don't want to take responsibility for it.

**Comment by BKPepe** [comment]
Then those patches should not be accepted in the first place.

**Comment by feckert** [comment]
@Ansuel @BKPepe I have now gone through all the commits again.
As discussed, adding a signed-off-by from the one who presumably created it is not desirable.
Also in the `From:` patch header the presumed creator, by searching the commit history is also not clear.

So I decided that at least we have a valid patch to replace the `From: ` Patch header with a dummy line `From: Unknown Person <Unknown.Person@unknown.domain.world>`this should solve the reason for the pullrequest.

As a follow up, I will search the history and post a summary on the mailing list asking the creators to please add a signed-off-by. Or find someone who knows what the patch does and feels responsible.

I hope this is a good approach.

Next, I would create another pullrequest that removes the `git am` apply warning.
`.git/rebase-apply/patch:30: space before tab in indent`.
There are a few.

**Comment by BKPepe** [comment]
> the patch was never signed off so we can't really assume that... even putting the from can be problematic.

I'd say that these patches are problematic at all. Nobody knows what they are doing, why they are here, and why they can not be dropped. My colleague said it is a can of worms, and nobody wants to open it. :see_no_evil: :see_no_evil: 
Also, those patches do not meet [submission guidelines](https://openwrt.org/submitting-patches#submission_guidelines). Thus this put as to the beginning of this story and how we can improve now. I am afraid that reaching anyone would be too long. Putting now ``Signed-off-by`` could be problematic, but some projects are forcing users to agree with contributing guidelines, and thus then, they can add ``Signed-off-by`` backward. Not sure how OpenWrt handles this.

But the ``Signed-off-by`` is one thing. By looking in the history, we can find who added the patch and put him/her as author. 

TL;DR: I see minor improvements over here, but many things are missing (commit description, etc.). Those patches should be revisited once and for all.

> Patch header with a dummy line From: Unknown Person <Unknown.Person@unknown.domain.world>this should solve the reason for the pullrequest.

Well, it is better than putting yourself as an author, but instead of Unknown Person, we should be more clear on this one:

a) ``OpenWrt community <openwrt-devel@lists.openwrt.org>``
b) ``OpenWrt developers <openwrt-devel@lists.openwrt.org>``

From my point of view, there should be a valid address, and the mailing list is the way to go on this one.

**Comment by Ansuel** [comment]
@BKPepe true that we should have rejected them from the start but the situation is not that bad... if you check the patch they are really easy one... It's a mix of:
- Adding description to kconfig flag (to fix kernel warning? to make them selectable? for kmods?)
- Add flash ids
- Add usb ids
- Fix headers in some file (for musl?) this would be worth to investigate and probably can be dropped... We should really check why they were introduced in the first place
- One mvebu patch... should not be too easy to get sob @LGA1150 is busy but wonder if he can give us some help
- One patch for mtk bad block mgmt... mtk maintainers are active so a patch on the mailing list should be enough to get a sob...

Aside from the last 2 patch everything else is so simple that we should really ignore the hassle of asking maintainers and just regenerate the patch if they are still needed... I know it's not that good but I can't see how someone would complain of stealing a patch that just adds an additional description in a kconfig... considering we should have rejected and he didn't care in the first place to add correct from/sob we should have power to fix it ourself.

**Comment by Ansuel** [comment]
I missed the kconfig exit patch

this was the link to the mailing list

https://www.mail-archive.com/openwrt-devel@lists.openwrt.org/msg59501.html

@blocktrron can you provide sob and from tag?

**Comment by Ansuel** [comment]
> In this PR as it is touching a few patches yes, but if you take a look at folders `hack-5.10` or `hack-5.15`, it is a mess, though. Some patches there are here for a long time. :-(

yhea the nightmare of having lede patch that changed name or even svn patch ahahhaha

**Comment by feckert** [comment]
> a) `OpenWrt community <openwrt-devel@lists.openwrt.org>` b) `OpenWrt developers <openwrt-devel@lists.openwrt.org>`

@Ansuel I have now rebased everything.
I have now added a valid git header wherever necessary and the `From:` now also fits as suggested by @BKPepe with 
`From: OpenWrt developers <openwrt-devel@lists.openwrt.org>`

**Comment by Ansuel** [comment]
@BKPepe think we can merge this and at the same time open a big issue to track all these patch and discuss there?
@feckert can you send a mail to the mailing list about these patches?

**Comment by BKPepe** [comment]
> @BKPepe think we can merge this and at the same time open a big issue to track all these patch and discuss there?

Yes, please! Thanks.

**Comment by Ansuel** [comment]
Thanks, merged in master with 
8857387913 linux/generic/hack-5.15: add missing patch headers
2ed7f6cbb0 linux/generic/pending-5.15: add missing patch headers
a9573a029a target/linux/pending-5.15: rename patches

**Comment by Ansuel** [comment]
@BKPepe opened https://github.com/openwrt/openwrt/issues/10272 i'm adding as info as possible if you want we can continue the discussion there
